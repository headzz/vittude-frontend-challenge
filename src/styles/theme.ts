export const theme = {
  colors: {
    primary: '#FF6952',
    secondary: '#F26B5B',
    error: '#F63D15',
    dark: {
      100: '#777C88',
      200: '#2E3035',
      300: '#000000',
    },
    light: {
      100: '#FFFFFF',
      200: '#E7EAF4',
    },
  },
  font: {
    size: {
      normal: '1rem',
      small: '0.875rem',
      large: '2rem',
      extraLarge: '3rem',
    },
    type: { header: 'Inter', normal: 'Fira Sans' },
  },
  spacing: {
    extraSmall: '10px',
    small: '12px',
    normal: '16px',
    large: '24px',
    extraLarge: '48px',
  },
}
