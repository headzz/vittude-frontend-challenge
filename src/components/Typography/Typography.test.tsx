import { theme } from '@styles/theme'
import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { Typography } from '.'

describe('Typography', () => {
  it('renders default text', () => {
    render(
      <ThemeProvider theme={theme}>
        <Typography>Test</Typography>
      </ThemeProvider>,
    )

    const paragraph = screen.getByText(/test/i)

    expect(paragraph).toMatchSnapshot()
  })

  it('renders Inter font and large text', () => {
    render(
      <ThemeProvider theme={theme}>
        <Typography font="Inter" size="large">
          Test
        </Typography>
      </ThemeProvider>,
    )

    const paragraph = screen.getByText(/test/i)

    expect(paragraph).toMatchSnapshot()
  })

  it('renders bold and extra large text', () => {
    render(
      <ThemeProvider theme={theme}>
        <Typography bold size="extraLarge">
          Test
        </Typography>
      </ThemeProvider>,
    )

    const paragraph = screen.getByText(/test/i)

    expect(paragraph).toMatchSnapshot()
  })
  it('renders small text', () => {
    render(
      <ThemeProvider theme={theme}>
        <Typography size="small">Test</Typography>
      </ThemeProvider>,
    )

    const paragraph = screen.getByText(/test/i)

    expect(paragraph).toMatchSnapshot()
  })
})
