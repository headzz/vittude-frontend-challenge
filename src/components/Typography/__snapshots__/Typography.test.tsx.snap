// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders Inter font and large text 1`] = `
<p
  class="sc-bcPKhP dvolhf"
>
  Test
</p>
`;

exports[`Typography renders bold and extra large text 1`] = `
<p
  class="sc-bcPKhP hoITMT"
>
  Test
</p>
`;

exports[`Typography renders default text 1`] = `
<p
  class="sc-bcPKhP disYCL"
>
  Test
</p>
`;

exports[`Typography renders small text 1`] = `
<p
  class="sc-bcPKhP ivqPcO"
>
  Test
</p>
`;
